main.js

```diff
@@ -1,65 +1,73 @@
+      currentMessage = affirmations[affirmationsIndex];
+      message.innerText = currentMessage;
+      var singularAffirmation = affirmations.splice(affirmationsIndex, 1);
+      usedAffirmations.push(singularAffirmation[0]);
+      revealMessage();
+    } else {
+      currentMessage = affirmations[0];
+      message.innerText = currentMessage;
+      var singularAffirmation = affirmations.splice(0, 1);
+      usedAffirmations.push(singularAffirmation[0]);
+      affirmations = usedAffirmations;
+      usedAffirmations = [];
+      revealMessage();
+      theEnd();
+      hideMessage();
+      // manageMessagesDisplayed(affirmations, usedAffirmations);
+    }
+  } else if (radioSelection[1].checked) {
+    if (mantras.length > 1) {
+      currentMessage = mantras[mantrasIndex];
+      var singularMantra = mantras.splice(currentMessage, 1);
+      usedMantras.push(singularMantra[0]);
+      message.innerText = currentMessage;
+      revealMessage();
+    } else {
+      currentMessage = mantras[0];
+      var singularMantra = mantras.splice(currentMessage, 1);
+      usedMantras.push(singularMantra[0]);
+      message.innerText = currentMessage;
+      mantras = usedMantras;
+      usedMantras = [];
+      revealMessage();
+      theEnd();
+      hideMessage();
+      // manageMessagesDisplayed(mantras, usedMantras);
+    }
   } else {
-    currentMessage = mantras[usedMantra];
-    manageMessagesDisplayed(mantrasCopy, mantras);
-    mantrasCopy.splice(currentMessage, 1);
-    console.log('this is the mantras array', mantras, 'this is the current message',
-      currentMessage);
+    alert("Please select from one of the two options to recieve your message.");
   }
-
-  message.innerText = currentMessage;
-  revealMessage();
 }
-//how can I make this more dynamic with a for loop? (radio buttons are arrays?!; another document.selector?)
 
-function manageMessagesDisplayed(arrayCopy, main) {
-  if (!arrayCopy.length) {
-    // if (array === affirmationsCopy) {
-    //   affirmationsCopy = affirmations;
-    // } else {
-    //   mantrasCopy = mantras
-    // }
-    alert("You've reached the end of our avaliable uplifting messages. Press 'ok' if you'd like to go through them again!")
-    console.log(arrayCopy);
-    return arrayCopy = main;
-    hideMessage();
-  }
-  //how can I repopulate an empty array on the dom??
-  //how to prevent undefined from appearing?
+// function manageMessagesDisplayed(originalData, usedData) {
+//   originalData = usedData;
+//   usedData = [];
+//   alert("You've reached the end of our avaliable messages. Please click 'ok' to view them again!");
+// }
+//I don't understand why this isn't working. So I hard coded it instead. If you could please walk me through why the repopulation wasn't working inside of a function but would work when hard coded, I would greatly appreciate it.
+
+function theEnd() {
+  alert("You've reached the end of our avaliable messages. Please click 'ok' to view them again!");
```

Comment:
If your function is only has one line of code in it, you probably can just do that one line of code the place you call the function. I would put use alert itself instead of the theEnd function.